include/nbl/video/IDescriptorPool.h

```diff
@@ -29,12 +38,87 @@ class NBL_API IDescriptorPool : public core::IReferenceCounted, public IBackendO
             uint32_t count;
         };
 
-        explicit IDescriptorPool(core::smart_refctd_ptr<const ILogicalDevice>&& dev, uint32_t _maxSets) : IBackendObject(std::move(dev)), m_maxSets(_maxSets) {}
+        inline void createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
+        {
+            for (uint32_t i = 0u; i < count; ++i)
+            {
+                auto layout = core::smart_refctd_ptr<const IGPUDescriptorSetLayout>(layouts[i]);
+                output[i++] = createDescriptorSet(std::move(layout));
+            }
+        }
+        
+        core::smart_refctd_ptr<IGPUDescriptorSet> createDescriptorSet(core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& layout);
```

Comment:
it should be implemented the other way round, the `createDescriptorSet` should be an inline and specified in terms of `createDescriptorSets`

Reply:
Resolved.